warden-roboto/src/main/kotlin/AndroidAttestationChecker.kt

```diff
@@ -112,6 +114,31 @@ abstract class AndroidAttestationChecker(
 
     protected abstract val trustAnchors: Collection<PublicKey>
 
+    protected open fun ParsedAttestationRecord.verifyAttestationTime(verificationDate: Date = Date()) {
+        val createdAt =
+            teeEnforced().creationDateTime().getOrNull() ?: softwareEnforced().creationDateTime().getOrNull()
+        if (createdAt == null) throw AttestationValueException(
+            "Attestation statement creation time missing",
+            reason = AttestationValueException.Reason.TIME
+        )
+        val calendar = Calendar.getInstance()
+        calendar.time = verificationDate
+        calendar.add(Calendar.SECOND, attestationConfiguration.verificationSecondsOffset)
+        var checkTime = calendar.toInstant()
+        val difference = Duration.between(createdAt, checkTime)
+        if (difference.isNegative) throw AttestationValueException(
+            "Attestation statement creation time too far in the future: $createdAt, check time: $checkTime",
+            reason = AttestationValueException.Reason.TIME
+        )
+
+        if (difference > Duration.ofSeconds(attestationConfiguration.attestationStatementValiditySeconds.toLong())) throw AttestationValueException(
```

Comment:
Is this unit tested somewhere? Because I find it hard to read ... wouldn't it be better to operate on instants for time manipulation?

Reply:
right. this is only tested in WARDEN, not here. i have copied the tests over and adapted them. I also changed it to instant internally